src/Microsoft.Identity.Client/Internal/Requests/AuthenticationRequestParameters.cs

```diff
@@ -113,30 +113,49 @@ internal class AuthenticationRequestParameters
 
         public void LogState()
         {
-            StringBuilder builder = new StringBuilder(Environment.NewLine + "=== Request Data ===" +
-                                                      Environment.NewLine +
-                                                      "Authority Provided? - " + (Authority != null) +
-                                                      Environment.NewLine);
-            builder.AppendLine("Client Id - " + ClientId);
-            builder.AppendLine("Scopes - " + Scope?.AsSingleString());
-            builder.AppendLine("Redirect Uri - " + RedirectUri?.OriginalString);
-            builder.AppendLine("Validate Authority? - " + ValidateAuthority);
-            builder.AppendLine("LoginHint provided? - " + !string.IsNullOrEmpty(LoginHint));
-            builder.AppendLine("User provided? - " + (User != null));
-            var dict = MsalHelpers.ParseKeyValueList(ExtraQueryParameters, '&', true, RequestContext);
-            builder.AppendLine("Extra Query Params Keys (space separated) - " + dict.Keys.AsSingleString());
-            dict = MsalHelpers.ParseKeyValueList(ExtraQueryParameters, '&', true, RequestContext);
-            builder.AppendLine("Slice Parameters Keys(space separated) - " + dict.Keys.AsSingleString());
-#if DESKTOP || NETSTANDARD1_3
-            builder.AppendLine("Confidential Client? - " + (ClientCredential != null));
-            builder.AppendLine("Client Credential Request? - " + IsClientCredentialRequest);
-            if(IsClientCredentialRequest)
+            // Create Pii enabled string builder
+            if (Logger.PiiLoggingEnabled)
             {
-                builder.AppendLine("Client Certificate Provided? - " + (ClientCredential.Certificate != null));
+                StringBuilder builder = new StringBuilder(Environment.NewLine + "=== Request Data ===" +
+                                                          Environment.NewLine +
+                                                          "Authority Provided? - " + (Authority != null) +
+                                                          Environment.NewLine);
+                builder.AppendLine("Client Id - " + ClientId);
```

Comment:
whole blob is not PII. you can selectively remove information that is qualified to be PII. this is a critical piece of information that we need

Reply:
I removed the pii info for the non-pii log.